src/main/fc/mw.c

```diff
@@ -829,25 +819,12 @@ void taskMainPidLoopCheck(void)
 
     const uint32_t startTime = micros();
     while (true) {
-        if (gyroSyncCheckUpdate(&gyro) || ((currentDeltaTime + (micros() - previousTime)) >= (gyro.targetLooptime + GYRO_WATCHDOG_DELAY))) {
-            static uint8_t pidUpdateCountdown;
-
+        if (isGyroDataReady() || ((currentDeltaTime + (micros() - previousTime)) >= (targetPidLooptime + GYRO_WATCHDOG_DELAY))) {
             if (debugMode == DEBUG_PIDLOOP) {debug[0] = micros() - startTime;} // time spent busy waiting
-            if (runTaskMainSubprocesses) {
-                subTaskMainSubprocesses();
-                runTaskMainSubprocesses = false;
-            }
-
-            gyroUpdate();
 
-            if (pidUpdateCountdown) {
-                pidUpdateCountdown--;
-            } else {
-                pidUpdateCountdown = setPidUpdateCountDown();
-                subTaskPidController();
-                subTaskMotorUpdate();
-                runTaskMainSubprocesses = true;
-            }
+            subTaskMainSubprocesses();
```

Comment:
I'm a little unsure of this part, shouldn't `subTaskMainSubprocesses` run less frequently than `subTaskPidController` and `subTaskMotorUpdate`?